server/repository/src/db_diesel/assets/asset_log.rs

```diff
@@ -0,0 +1,205 @@
+        }
+
+        let final_query = query
+            .offset(pagination.offset as i64)
+            .limit(pagination.limit as i64);
+
+        // Debug diesel query
+        // println!(
+        //    "{}",
+        //     diesel::debug_query::<DBType, _>(&final_query).to_string()
+        // );
+
+        let result = final_query.load::<AssetLog>(&self.connection.connection)?;
+
+        Ok(result.into_iter().map(to_domain).collect())
+    }
+}
+
+fn to_domain(asset_log_row: AssetLogRow) -> AssetLog {
+    asset_log_row
+}
+
+type BoxedAssetLogQuery = IntoBoxed<'static, asset_log::table, DBType>;
+
+fn create_filtered_query(filter: Option<AssetLogFilter>) -> BoxedAssetLogQuery {
+    let mut query = asset_log_dsl::asset_log.into_boxed();
+
+    if let Some(f) = filter {
+        let AssetLogFilter {
+            id,
+            asset_id,
+            status,
+            log_datetime,
+        } = f;
+
+        apply_equal_filter!(query, id, asset_log_dsl::id);
+        apply_string_filter!(query, status, asset_log_dsl::status);
+        apply_date_filter!(query, log_datetime, asset_log_dsl::log_datetime);
+
+        if let Some(asset_id) = asset_id {
+            let mut sub_query = asset_dsl::asset.select(asset_dsl::id).into_boxed();
+            apply_equal_filter!(sub_query, Some(asset_id), asset_dsl::id);
+            query = query.filter(asset_log_dsl::asset_id.eq_any(sub_query));
+        }
+    }
+    query
+}
+
+#[cfg(test)]
+
+mod tests {
+    use crate::{
+        assets::asset_log::{AssetLogFilter, AssetLogRepository},
+        mock::MockDataInserts,
+        test_db, EqualFilter, Pagination,
+    };
+
+    use super::{AssetLogSort, AssetLogSortField};
+
+    #[actix_rt::test]
+    async fn test_asset_log_query_repository() {
+        let (_, storage_connection, _, _) = test_db::setup_all(
+            "test_asset_log_sort_query_repository",
+            MockDataInserts::none().assets().asset_logs(),
+        )
+        .await;
+        let asset_log_repository = AssetLogRepository::new(&storage_connection);
+
+        // test query by one
+
+        let asset_log_id = "log_a".to_string();
+        let log = asset_log_repository
+            .query_one(AssetLogFilter::new().id(EqualFilter::equal_to(&asset_log_id)))
+            .unwrap()
+            .unwrap();
+
+        assert_eq!(log.id, asset_log_id);
+
+        // test query multiple
+
+        let logs = asset_log_repository
+            .query(Pagination::new(), None, None)
+            .unwrap();
+
+        assert_eq!(logs.len(), 3);
+
+        // test query multiple with sort
+
+        let logs_sorted_by_datetime = asset_log_repository
+            .query(
+                Pagination::new(),
+                None,
+                Some(AssetLogSort {
+                    key: AssetLogSortField::LogDatetime,
+                    desc: None,
+                }),
+            )
+            .unwrap();
+
+        assert_eq!(logs[0], logs_sorted_by_datetime[1]);
```

Comment:
```suggestion
         assert_eq!(logs_sorted_by_datetime[0], mock_asset_log_c());
```
You haven't got any sort parameter for your `logs` query, so the database might return the data in any order.
This checks that the earliest datetime in the mocks is correct.